### ISSUE TITLE
Add default implemention of build for Buildable 

### ISSUE THREAD

**Issue by chshersh** [cdornan/fmt#10]
Very often you want to write just 

```haskell
instance Buildable MyType where
    build = genericF
```

It may be better to have default implementation of `build` (which is not possible, because we don't use our own `Buildable`). But still things should be easier...

**Comment by neongreen** [comment]
I'm starting to think we should just use our own `Buildable`, with a fallback to `Formatting.Buildable` if there's no instance.

**Comment by neongreen** [comment]
Or even a fallback to `Show`.

**Comment by dcastro** [comment]
An alternative solution that wouldn't require having our own `Buildable` would be to leverage `DerivingVia`, e.g.:

```hs
newtype GenericBuildable a = GenericBuildable a

instance (GBuildable (Rep a), Generic a) => Buildable (GenericBuildable a) where
  build (GenericBuildable a) = genericF a
```

Usage:

```hs
data D = D
  deriving stock Generic
  deriving Buildable via GenericBuildable D
```

@neongreen @cdornan what do you think? I'd be happy to create a PR.

**Comment by cdornan** [comment]
@dcastro I like the `DerivingVia` idea : nondisruptive and really gets the job done. Feel free to raise a PR. (Sorry for the delay &mdash; other things have been sucking up available bandwidth.)

**Comment by dcastro** [comment]
@cdornan Thanks for getting back to me! No need to apologise, OSS is draining and life is wild, I completely understand :smile: I'll try to raise a PR in the coming weeks then, cheers!

**Comment by cdornan** [comment]
Fixed by #38
